### DISCUSSION TITLE
Many printer.cfg issues.

### DISCUSSION THREAD

**Discussion by Astrolava** [bassamanator/Sovol-SV06-firmware#22] [Q&A]
I've been trying to get my sv06 working with the SKR mini E3 V3, and so far I've managed to make some of it work.
The printer.cfg on the bassamanator's main page doesn't play nice with the autohoming, while the printer.cfg listed in the "issues" section doesn't play nice with the extruder / heatbed pins.
I've had to pick and choose snippets from each to get it working with my printer, however, my benchy has an almost melting effect.  Overhangs are just failing, which seems to indicate overheating.  I didn't change my slicer settings, so I wonder if the temperature is higher than what I've configured it for.

I'm printing with PETG, 250c, fan at 0%, 20% after layer 3.  Bed temp 85c.
These were optimal settings on the stock board using marlin.  I'm not sure why it's failing me now.

![benchy](https://user-images.githubusercontent.com/45184363/230222558-5e908662-2dca-4177-9d39-0c7d4b7e12d0.jpg)

This is my printer.cfg.  I had to convert it to .txt because of github.

[printer.txt](https://github.com/bassamanator/Sovol-SV06-firmware/files/11163451/printer.txt)

Any ideas/advice/help is highly appreciated.

**Comment by bassamanator** [comment]
> The printer.cfg on the bassamanator's main page doesn't play nice with the autohoming

The `printer.cfg` on the `master` branch is for the stock SV06, not for the skr.

>while the printer.cfg listed in the "issues" section doesn't play nice with the extruder / heatbed pins.

As clearly stated, that is for the skr-mini-e3-v2. And it's in the issues section, you shouldn't be using that.

Whatever `printer.cfg` you're using, it seems to be working perfectly from what I see, given that you were able to print an entire benchy. And 20% fan speed basically means that the fan is off. You can't expect overhangs to come out nice if your fan is off.

I print my PETG at 100% fan speed. Many of [these](https://github.com/bassamanator/everything-sovol-sv06/tree/main/images/prints) prints are PETG.

**Reply by bassamanator** [reply]
So basically, you've managed to get everything working correctly, simply increase fan speed.

**Comment by Astrolava** [comment]
Sorry for the confusion.  I was using the configuration from this branch:
https://github.com/bassamanator/Sovol-SV06-firmware/tree/skr-mini-e3-v3

When I used this as it was, the autohome wouldn't work.  I tried to diagnose the issue through discord, and there was a lot of concern addressed towards the "run current" being "quite high".  Of course, I don't know anything about that.
Using some snippets from the V2 fixed my autohome issue, though I don't know why.  It was more or less a shot in the dark for me, but as the previous configuration didn't work, I thought it was worth at least trying since the autohome was absolutely required for my printer to be operational.  To better clarify, the extruder and bed would not move at all during autohome, but would move to the "center" relative to the position they were already in, sometimes leading to them hitting their limit and then skipping horribly until I killed the power.
I can revert back to the exact configuration from the branch linked above and try to iron it out.  I'm pretty sure I wired everything correctly.  I followed the diagram intensively.  
I would like to note that my PSU is switched to 110v as I'm in Japan.  I'm not sure if that makes a difference to the run currents or fan percentages.  I'm no electrician.

It was mentioned that my print appeared to be printing "too hot" and some concern was raised as to which "thermistor table" I used while configuring.  I've been using Klipper for maybe a day, and have no idea about that.
I had good prints using marlin and the stock motherboard under these exact slicer settings, including the fan percentage, so I'm just perhaps lacking the understanding as to why there would be such a variation in quality.
I will attempt a benchy with the fan at 100% as per your suggestion.

If there's anything I may be able to contribute to help alleviate potential issues for people in similar situations, please let me know.
I've saved my current configuration as a backup, but I really think I should use the proper configuration.

**Reply by bassamanator** [reply]
Thanks for the clarifications. I hope I didn't come off as too harsh. If I did, then my appologies.

First thing that I want you to do is to reprint the benchy, but increase fan speed to 100%. You also have to realize, you might be printing faster than you were on Marlin, so fan speed comes into play; if you're printing faster, you better cool down the previous layer before you lay down the next layer. Again, that benchy looks spot on to me except for the overhangs, so whatever you've done, you've done it right.

People usually like to comment on matters that they don't understand (I know that this was simply the users that you found on discord, and are not your statements). The `run_current` are not "quite high", though perhaps the z motor currents are on the "higher side" of things. The `run_current` values are pulled from the Marlin settings, **Sovol3d themselves have set these values**, they have nothing to do with me. I completely trust Sovol3d in this matter, they don't want the steppers blowing up on their customers, and have set these currents based on the spec sheet of the steppers. After 244h of print time, I have encountered no issues with skipping/overheating, etc. As a comparison, the fancy LDO motors on my Voron 0.1 have `run_current` values of `0.78` for the x and y steppers, these are according to the specs. Compare this value to the no-name steppers on the SV06 (0.86 and 0.9), and you see that they are well within an acceptable range. You can always try to reduce current and stop as soon as you experience skipping.

The psu voltage 110/220 doesn't play a role here btw, just set that figure according to where you live; I'm on 110V myself.

BTW you can always reverse the direction of the stepper movement by adding a `!` or removing it, from `dir_pin` property. `dir_pin: PC5` vs `dir_pin: !PC5`.

Although I do have an skr-mini-e3-v3 on the way, it's going to take some time. (I don't plan on switching really, will probably sell it, there was a sale on aliexpress and I got it for an unbeatable price.) I would love your help in getting this printer.cfg up to speed so that other users have an easier time.

PS anytime you're developing a `printer.cfg`, your finger should be on the PSU switch. An `M112` in the klipper console (emergency stop) acts even faster than cutting off the power.

**Comment by Astrolava** [comment]
After a few prints and lots of adjusting, I got almost a benchy.  So many issues.  One print also failed, stating it was "too far in the past" or something, so I had to adjust the microsteps to 32 across the board.
I'm still adjusting, seeing what's a slicer config issue and what's a printer.cfg issue.  I've changed the run_currents to .86, .9, and 1.0, and the extruder to .550.
I believe my printer.cfg is fully functional at this point.  I'm a little confused about that mcu error regarding the microsteps, and thought maybe it's because I added interpolation to the y axis (the bed became too loud and it was keeping me awake)
I've since turned off the interpolation on the y axis as well.
Stealthchop is 999999 though.  I think that's less of a printer.cfg issue and more of a preference, though I do wonder what the original marlin firmware's setting was in that regard, as the bed was quite quiet while running that firmware.

I'm printing yet another benchy.  If I run into any more mcu errors, I'll be sure to post that here.

**Comment by bassamanator** [comment]
>One print also failed, stating it was "too far in the past" or something, so I had to adjust the microsteps to 32 across the board.

The microsteps should not have anything to do with this, but okay. BTW I hope you were restarting the firmware (firmware_restart) after every change to `printer.cfg`.

`interpolate: True` reduces stepper motor accuracy, just so you know. Unless you have a special use case, you should set that to `False`.

Setting `microsteps` to a higher value is known (I've noticed this myself) to reduce stepper motor whine, as well as improving skin texture. I would recommend `64` but please use a value based on your preference.

Do you mind posting your `printer.cfg`, I want to do a comparison.

**Comment by Astrolava** [comment]
Sure, sorry about the delay.
I've had to convert it to txt as github doesn't support cfg.  
I haven't got around to changing the microsteps back to 64.
I was restarting my firmware each time I made a change to printer.cfg, however, I know for some changes a more thorough restart is recommended.  Maybe my lack of a thorough restart caused the issue?  I'll change the microsteps to 64 and try printing another benchy.
[printer.txt](https://github.com/bassamanator/Sovol-SV06-firmware/files/11183625/printer.txt)

**Reply by bassamanator** [reply]
For any changes to `printer.cfg`, a simple `FIRMWARE_RESTART` is sufficient.

I did notice that your `[bed_screws]` section is incorrect, but not relevant at the moment.

The only difference I found between our configs is that for `stepper_x`, you have `dir_pin: PB12` and I have `dir_pin: !PB12`. I will make that adjustment.

Your `printer.cfg` is irrelevant at this point because it is correct.

Do you happen to have any PLA on hand? It is the easier plastic to print with. The slicer in use is not relevant here, however, just to remove 1 more variable, please use SuperSlicer (official release) and use the config bundle included on this repo.

Keep this discussion going, it might be relevant to other users.

**Comment by bassamanator** [comment]
@Astrolava Also noticed your sensitivity levels are very sensitive, your prints looking the way they do make sense now. I discovered this while fine tuning the sensitivity values.

**Reply by Astrolava** [reply]
Do you know specifically what I need to adjust?  I'm worried about making changes, as this is my first printer and I'm actually not even sure what most of the settings do.

**Comment by bassamanator** [comment]
Change to `driver_SGTHRS: 81` everywhere.

Also check [closed issue#49](https://github.com/bassamanator/Sovol-SV06-firmware/issues/49). Someone setup there printer with the skr mini e3 v3.

EDIT: I would recommend you pull the skr branch again and run through the setup except you don't need to re-flash.

**Reply by superceleron** [reply]
im using a skr 3 ez on mine, i did had to lower the driver_SGTHRS to 70 and the z's run_current to 0.700 because at 1.000 they get burning hot and i mean really burning that you smell it!

**Reply by bassamanator** [reply]
@superceleron Could you please link to the `printer.cfg` for the skr 3 ez? I'd like to create a branch for it here. Will give you full credit for it of course and will link to your github profile.

**Reply by superceleron** [reply]
Ye no problem, i will do it tomorrow.
Mine is mix of yours + this one (https://github.com/Pr20100/SOVOL-SV06-Klipper-profile/).
But i modified it a lot now, i added Z_Align + Kamp + the display works + Filament runout +fans etc etc... all pins for the SKR 3 EZ fill in.
In mine on the drivers.cfg you going to notice i use different "run_current" for the both Z motors, i did that because my left one at 0.700 gets really hot, and the right one works at a "normal" temp (60/70c this to the touch... so not that accurate), i could not do both at 0.600 because if i do both run a different speeds.. strange i know, but i think this motors quality are not that great, so i use the left at 0.600 and the right one at 0.700 and they run both at the correct speed ( like i said strange.....).
Also im using the microsteps at 256 at the steppers, and man it works great!

**Reply by bassamanator** [reply]
Oh I see. Well at least steppers are easy enough to find, but yeah, it's about $20 a pop.

Interestingly enough the stock board can handle `microsteps: 256` *most* of the time. It's just that for some movements the mcu is unable to generate the microsteps and crashes. What the frequency of that ez board? 72 MHz stock, and 64 MHz skr-mini-e3-v2.0. 

What's the Z_align that you mentioned?

**Reply by superceleron** [reply]
This chip is 550Mhz but is running at 400Mhz.
The Z_align is a method you do not need to use the gantry smashing on top, at start_print is does the z_align(gantry alignment) and works great, since for me the gantry smashing on top never did a good job the left side was always a bit higher that the right side.
So it works like this, it measures with the prob to opposite sides of the bed (in this case i use the middle of the bed and then the opposite sides the middle of the bed), and auto aligns the gantry to the bed of course for that you need the 5 steppers, so you have the independent Z motors.
Btw, the Z motors are 0.800 max and the xy and 1.3.

**Reply by bassamanator** [reply]
550/400Mhz? Yowzers! 

Ah then the board must have 2 drivers for the z steppers. Nice!

**Reply by superceleron** [reply]
well not 2 drivers for Z, it as 2 extruder drivers, but you can use E1 for one of the Z :) so ye same thing lol.

<img width="672" alt="image" src="https://github.com/bassamanator/Sovol-SV06-firmware/assets/11303825/4339b41f-ad0f-443b-955a-09b071162941">

**Reply by superceleron** [reply]
https://github.com/superceleron/SV06_SKR_3_EZ_Klipper

**Comment by bassamanator** [comment]
What timezone are you in?

**Reply by Astrolava** [reply]
I'm in Japan.

**Reply by bassamanator** [reply]
Do you have time to work on your printer right now?

**Reply by Astrolava** [reply]
yes

**Reply by bassamanator** [reply]
https://talky.io/beneath-a-dry-water

**Reply by Astrolava** [reply]
Should I update Klipper?  We just refreshed everything, but it looks like there are a couple updates.  I'm not sure if the updates will revert any of the desired changes.

**Reply by bassamanator** [reply]
Yes use mainsail to update everything. You might not be able to find the mcu after the updates, in case that happens, locate the mcu via `ls -l /dev/serial/by-path/` instead of the `by-id` used earlier.

**Reply by bassamanator** [reply]
Another thing that you might want to do is to reflash the board via the new `.bin` that I uploaded earlier. Warning, it has not been tested.

**Reply by Astrolava** [reply]
Yeah those steppers got REALLY hot.  I touched one to check and nearly got a burn.  I'll adjust accordingly.

**Reply by bassamanator** [reply]
Change x, y, z `run_current: 0.580`.

EDIT: these values might need further adjustment.

**Reply by bassamanator** [reply]
Before the changes today, did you happen to notice stepper temperatures?

**Reply by Astrolava** [reply]
I felt them occasionally, but never managed to measure the actual temperature.  They were warm, but never hot.  I'd say maybe around 30-40 degrees at most.

**Reply by bassamanator** [reply]
Got it.

Please use the superslicer config bundle included on the repo, **just for the test prints** (official release only). This way I can gauge who to blame for print quality 🤣.

**Reply by bassamanator** [reply]
@Astrolava Any updates?

**Reply by Astrolava** [reply]
My ribbon cable broke and now I need a new one. :(

**Reply by Astrolava** [reply]
Just an update.  I replaced the cable.  Sovol sent me a new one.  I printed some parts for an ERCF and the quality doesn't look perfect, but it's passable.  I think all is well currently.

**Reply by bassamanator** [reply]
ERCF is a carbon fiber of some sort? How about regular PLA/+, how does that print?

**Reply by Astrolava** [reply]
ERCF is a multicolor thing, I'm printing it in PETG.

**Comment by Astrolava** [comment]
Hey, I know it's been a while, but I was working with input shaper and figured I'd give another crack at setting the run currents.  
I came across 3 links, https://sv06.blakadder.com/Parts/mechanical-parts/#z-axis-couplers This link gives all the info of the parts of the SV06.
This https://www.youmaketech.com/wp-content/uploads/2022/01/Creality-motors.jpg gives the Max current.
And finally, this gives the run current: 
https://docs.vorondesign.com/community/howto/120decibell/calculating_driver_current.html
When I do all that, I get a run current around 0.59, which is far different from the 0.860 it was originally set at.  I did set it to 0.580 on a whim as I didn't like hot stepper motors, but now I think maybe that's around where they were supposed to be all along.  Can you check these sources and confirm?

**Comment by bassamanator** [comment]
>  https://sv06.blakadder.com/Parts/mechanical-parts/#z-axis-couplers

You could have gotten that info from my other repo: [everything-sovol-sv06](https://github.com/bassamanator/everything-sovol-sv06#everything-sovol-sv06).

You can trust vorondesign.com to be quite accurate.

With the help of another user, I had determined that 0.600 would be a good value for the steppers, it's been like that for a while now. I'm glad we weren't far off from the official calculations. As I mentioned before (see above), the currents were pulled from the official Sovol firmware, but somehow only makes the steppers connected to the skr-mini-e3 hot.

I may even try reducing the currents on my steppers, just to see what happens.

**Comment by Astrolava** [comment]
Cool, I'll switch them all over and continue with input shaper.  It is strange how different boards seem to handle different currents differently.  
I wonder if maybe there was a conversion issue somewhere down the production line.

**Comment by Astrolava** [comment]
hey @bassamanator, sorry, I don't actually know how to contact you outside of this particular github forum.  
I noticed a couple days ago you posted an image of an amonite dual 5015 fan duct you printed.  I am also using this fan duct, but I've noticed even at 100% fan speed, I'm not getting the cooling I expected.  I still have warping bows on my benchies, blobs, and occasionally my overhangs still droop.  Now, this could be because I'm using some crappy petg, it might be possible I'm printing too hot, too fast, or too slow.  honestly I have no idea.
 I saw a better fan duct on cults3D for around 6 or 7 dollars, and I believe you mentioned it once before saying there's no better fan duct.
 however, it's a single 5015, where a dual 5015 does seem to have more potential.
Do you think a single 5015 with the cults3D fan is better than a dual 5015 using amonite?  
I don't have access to the cults3D design, as well, printing it in ASA is a pain.  I thought perhaps you had experience using both designs.  
Sorry for using this medium to contact you, but I haven't really seen anyone else mention much about fan ducts.
Some said the amonite is a bad duct design, but failed to provide any alternatives.  I do feel like maybe if one of the fans were spun upside down so both openings faced opposite of each other - it might have more intake.
Please let me know your thoughts, and maybe we can move that discussion somewhere else.  Thanks.

**Comment by bassamanator** [comment]
Messaging me here works best.

Check the useful parts section on my everything sv06 repo.

I use the cults3d model and it's excellent. Fairly easy to print and quite durable in PETG.

I have printed the ammonite but simply to examine the model, it's completely untested. It is supposed to be CFD optimized.

There are too many variables at play here. You should use a good PETG, Eryone is my favourite, and you should use good print settings.

BTW PETG doesn't need much cooling at all, you're fine with 30% with a single 5015 fan, so that might be something to think about.
